internal/controller/platform/services/services.go

```diff
@@ -105,12 +109,41 @@ type PlatformServiceHandler interface {
 
 	// Check if K_SINK has injected for Job Service. No Op for Data Index
 	CheckKSinkInjected() (bool, error)
+
+	// Returns whether job based, service based or no DB migration is needed
+	GetDBMigrationApproach() operatorapi.DBMigrationStrategyType
+	// Returns true if job based db migration, false otherwise
+	IsJobsBasedDBMigration() bool
+	// Returns true if service based db migration, false otherwise
+	IsServiceBasedDBMigration() bool
+	// Returns true if no DB migration is needed
+	IsNoDBMigration() bool
```

Comment:
I don't think this is necessary. When needed, just do the `GetDBMigrationStrategy() == strategy`.